justfile

```diff
@@ -69,3 +70,11 @@ install-toolchain:
 
 # The following line gives highlighting on vim
 # vim: set ft=make :
+
+benchmark firmware=default:
+	mkdir -p {{benchmark_folder}}
+	cargo run --package runner -- run -v --firmware {{firmware}} --benchmark --bench-type=instr\
+	| grep instr | cut -c 59- > {{benchmark_folder}}/instr.bench
```

Comment:
Would those commands work on windows? We don't really support windows but I's good to keep in mind and try to limit the number of platform-specific tools we need to use.

Reply:
It will not works. I still need to find a way to get the output of Miralis properly. As the benchmark code is running in Miralis, it doesn't have access to std function and cannot write to a file by itself.

I was wondering how we can "wrap" the execution of miralis inside another code that have access to std function. Maybe in order to merge `benchmark.rs` into `benchmark-analyzer` at the end.

Reply:
So the way I would do it is to use the runner for this. When you create a command you can redirect the stdin/stdout/stderr however you like. So the runner could capture the stdout stream, filter the lines with the appropriate tag and process them directly from the runner. So basically merging the runner and your new benchmark analyzer.

It seems a process can return an [Output](https://doc.rust-lang.org/std/process/struct.Output.html) in Rust, which collects the stdout as a `Vec<u8>`, maybe that is what you need?

Reply:
I can use the `output()` function on `qemu_cmd` to collect stdin and stderr and then write it into a file.

I think I'll directly merge benchmarking into the runner so that it can be executed directly from the collected output and avoid use of a file.